### ISSUE TITLE
Support execution graph capture via stream_t's

### ISSUE THREAD

**Issue by eyalroz** [eyalroz/cuda-api-wrappers#458]
CUDA execution graph templates can be created in one of two ways: Explicit construction and capture of operations enqueued via a stream. I'm currently working on the explicit construction API (see the [graph_support](https://github.com/eyalroz/cuda-api-wrappers/tree/graph_support) branch), but I should also add the support for capturing, to the stream wrappers.

```
CUresult cuStreamBeginCapture ( CUstream hStream, CUstreamCaptureMode mode );
CUresult cuStreamEndCapture ( CUstream hStream, CUgraph* phGraph );
CUresult cuStreamGetCaptureInfo ( CUstream hStream, CUstreamCaptureStatus* captureStatus_out, cuuint64_t* id_out );
CUresult cuStreamGetCaptureInfo_v2 ( CUstream hStream, CUstreamCaptureStatus* captureStatus_out, cuuint64_t* id_out, CUgraph* graph_out, const CUgraphNode** dependencies_out, size_t* numDependencies_out );
CUresult cuStreamIsCapturing ( CUstream hStream, CUstreamCaptureStatus* captureStatus );
CUresult cuStreamUpdateCaptureDependencies ( CUstream hStream, CUgraphNode* dependencies, size_t numDependencies, unsigned int  flags );
CUresult cuThreadExchangeStreamCaptureMode ( CUstreamCaptureMode* mode );
```

**Comment by eyalroz** [comment]
begin and end capture now seem to be working, IsCapturing implemented but not checked, need to add support for GetCaptureInfo (both of them?), UpdateCaptureDependencies, ExchangeStreamCaptureMode.

**Comment by raplonu** [comment]
Hi, 

I started using your graph API. The design is good, although I noted few issues.

Using c++20 and `std::span` the project does not compile due to incorrect usage. I tried to fix this issue in #478.
Let me know if you want me to modify it.

In addition, it would be nice if `instance_t` and `template_t` both have move assignment operator.

**Comment by eyalroz** [comment]
Yes, I forgot to add move assignment to them both.  Added now.
